sale_order_line_input/views/sale_order_line_view.xml

```diff
@@ -36,8 +36,7 @@
                 />
                 <field
                     name="order_partner_id"
-                    domain="[('customer','=',True)]"
-                    context="{'search_default_customer':1, 'show_address': 1}"
+                    context="{'res_partner_search_mode': 'customer', 'show_address': 1}"
```

Comment:
```suggestion
                    context="{'search_default_customer': 1,'res_partner_search_mode': 'customer', 'default_is_company': True, 'default_customer_rank': 1}"
```
We should use the same context that the customers action uses:
https://github.com/odoo/odoo/blob/4b1a9b2868b024ac3dc3a416f490a68ae09b7379/addons/account/views/partner_view.xml#L233

Reply:
Note that sales order view doesn't do that, which may be consider a bug:

https://github.com/odoo/odoo/blob/4b1a9b2868b024ac3dc3a416f490a68ae09b7379/addons/sale/views/sale_views.xml#L310